src/Raven.Server/Documents/Handlers/Processors/Subscriptions/AbstractSubscriptionsHandlerProcessorForCreateOrUpdateSubscription.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.TrafficWatch;
+using Raven.Server.Web;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Processors.Subscriptions
+{
+    internal abstract class AbstractSubscriptionsHandlerProcessorForCreateOrUpdateSubscription<TRequestHandler, TOperationContext> : AbstractDatabaseHandlerProcessor<TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractSubscriptionsHandlerProcessorForCreateOrUpdateSubscription([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected abstract bool GetDisabled(SubscriptionCreationOptions options);
+
+        protected abstract long? GetSubscriptionId(SubscriptionUpdateOptions options);
+
+        protected abstract ValueTask CreateInternalAsync(SubscriptionUpdateOptions options, TransactionOperationContext context);
+
+        public override async ValueTask ExecuteAsync()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
```

Comment:
why do you need to open a read tx here?

Reply:
its used to read from storage in `CreateSubscriptionInternalAsync`